src/invalid.ts

```diff
@@ -0,0 +1,77 @@
+import {SignalRef} from 'vega';
+import {ScaleChannel} from './channel';
+import {Mark, MarkDef} from './mark';
+import {isObject} from 'vega-util';
+
+/**
+ * Mixins for Vega-Lite Spec's Mark Definiton (to add mark.invalid)
+ */
+export interface MarkInvalidMixins {
+  /**
+   * Invalid data mode for marks, which defines how the visualization should represent invalid values (`null` and `NaN` in continuous scales without defined output for invalid values) in the marks and their scale domains.
+   *
+   * - `"filter"` — *Exclude* all invalid values from the visualization's *marks* and *scales*.
+   * For path marks (for line, area, trail), this option will create paths that connect valid points, as if the data rows with invalid values do not exist.
+   *
+   * - `"break-paths-filter-domains"` —
+   * Break path marks (for line, area, trail) at invalid values.  For non-path marks, this is equivalent to `"filter"`.
+   * All *scale* domains will *exclude* these filtered data points.
+   *
+   * - `"break-paths-show-domains"` —
+   * Break paths (for line, area, trail) at invalid values.  Hide invalid values for non-path marks.
+   * All *scale* domains will *include* these filtered data points.
+   *
+   * - `"show"` or `null` —
+   * Show all data points in the marks and scale domains. Each scale will use the output for invalid values defined in `config.scale.invalid`
+   * or, if unspecified, by default invalid values will produce the same visual values as zero (if the scale includes zero) or the minimum value (if the scale does not include zero).
+   *
+   * - `"break-paths-and-show-path-domains"` (default) —
+   * This is equivalent to `"break-path-keep-domains"` for path-based marks (line/area/trail)
+   * and `"filter"` for other marks.
+   *
+   * __Note__: If any channel's scale has an output for invalid values defined in `config.scale.invalid`,
+   * all values for the scales will be considered "valid" since they can produce a reasonable output for the scales.
+   * Thus, fields for such channels will not be filtered and will not cause path breaks.
+   */
+  invalid?: MarkInvalidDataMode | null;
+}
+
+export type MarkInvalidDataMode =
+  | 'filter'
+  | 'break-paths-filter-domains'
+  | 'break-paths-show-domains'
+  | 'break-paths-and-show-path-domains'
```

Comment:
Why is this the only one that includes the word `and`? Several of these options feel like conjunctive payloads, but this one is special for some reason?

Reply:
good call! fixed.